Rocket.Chat/Views/Cells/Chat/ChatMessageImageView.swift

```diff
@@ -43,22 +46,64 @@ final class ChatMessageImageView: UIView {
         return UITapGestureRecognizer(target: self, action: #selector(didTapView))
     }()
 
+    private func checkHttpImage(imageURL: URL) {
+        if imageURL.absoluteString.starts(with: "http://") {
+            isLoadable = false
+            labelTitle.text = attachment.title + " (" + localized("alert.insecure_image.title") + ")"
+            placeholderImage = UIImage(named: "Resource Unavailable")
+            imageView.contentMode = UIViewContentMode.center
+        }
+    }/*
+
+    private func openImageExternal(imageURL: URL) {
+        let controller = SFSafariViewController(url: imageURL)
+        present(controller, animated: true)
+    }*/
+
     fileprivate func updateMessageInformation() {
         let containsGesture = gestureRecognizers?.contains(tapGesture) ?? false
         if !containsGesture {
             addGestureRecognizer(tapGesture)
         }
 
+        guard let imageURL = attachment.fullImageURL() else {
+            return
+        }
+
         labelTitle.text = attachment.title
+        checkHttpImage(imageURL: imageURL)
 
-        let imageURL = attachment.fullImageURL()
         activityIndicatorImageView.startAnimating()
-        imageView.sd_setImage(with: imageURL, completed: { [weak self] _, _, _, _ in
+        imageView.sd_setImage(with: imageURL, placeholderImage: placeholderImage, completed: { [weak self] _, _, _, _ in
             self?.activityIndicatorImageView.stopAnimating()
         })
     }
 
     @objc func didTapView() {
-        delegate?.openImageFromCell(attachment: attachment, thumbnail: imageView)
+        if isLoadable {
+            delegate?.openImageFromCell(attachment: attachment, thumbnail: imageView)
+        } else {
+            guard let imageURL = attachment.fullImageURL() else {
+                return
+            }
+            ChatViewController.shared?.ask(key: "alert.insecure_image", buttonA: localized("global.ok"), buttonB: localized("chat.message.open_browser"), callbackB: { _ in
+                ChatViewController.shared?.openURL(url: imageURL)
+            })
+        }
+    }
+}
+
+extension ChatViewController {
+    func ask(key: String, buttonA: String, buttonB: String, callbackA: ((UIAlertAction) -> Swift.Void)? = nil, callbackB: ((UIAlertAction) -> Swift.Void)? = nil) {
```

Comment:
@cardoso in favour of #1201 - move this to Helpers/Ask.swift?

Reply:
Why do we need another method that implements the alert? Could we use the same one we already have implemented? :-)

Reply:
@rafaelks haven't seen anything with a settable callback, just the `Alert()`

Reply:
I see... but I guess we can improve that method, because I'm sure other parts of the code could take advantage of it. Let's make it more "generic". If you feel like not doing that right now, could you just use the regular UIAlertController for now and then we optimize later, please?